code/game/objects/structures/stool_bed_chair_nest/chairs.dm

```diff
@@ -396,6 +406,130 @@
 		span_warning("You repair \the [src]."))
 		chair_state = DROPSHIP_CHAIR_FOLDED
 
+/obj/structure/bed/chair/dropship/doublewide
+	name = "doublewide seat"
+	icon_state = "doublewide_chair" //only facing south cause the rest are ugly
+	max_integrity = 130
+	var/chair_color = NO_CHAIR_COLOR
+	var/leader_chair = FALSE
+	/// pixel x shift to give to the buckled mob
+	var/buckling_x = 0
+
+/obj/structure/bed/chair/dropship/doublewide/attack_alien(mob/living/carbon/xenomorph/X, damage_amount = X.xeno_caste.melee_damage, damage_type = BRUTE, damage_flag = "", effects = TRUE, armor_penetration = 0, isrightclick = FALSE)
+	if(X.status_flags & INCORPOREAL)
+		return FALSE
+	if(LAZYLEN(buckled_mobs))
+		unbuckle_mob(buckled_mobs[1])
+	return ..()
+
+/obj/structure/bed/chair/dropship/doublewide/welder_act(mob/living/user, obj/item/I)
+	welder_repair_act(user, I, 130, 1 SECONDS, 0, SKILL_ENGINEER_METAL, 1)
+	chair_state = DROPSHIP_CHAIR_FOLDED
+	update_icon()
+	return
+
+/obj/structure/bed/chair/dropship/doublewide/update_icon_state()
+	. = ..()
+	var/percentage = (obj_integrity / max_integrity) * 100
+	switch(percentage)
+		if(-INFINITY to 65)
+			chair_state = DROPSHIP_CHAIR_BROKEN
+			if(LAZYLEN(buckled_mobs))
+				unbuckle_mob(buckled_mobs[1])
+				balloon_alert_to_viewers("This chair is too damaged to sit in")
+	icon_state = "doublewide_chair[chair_color][chair_state]"
+
+/obj/structure/bed/chair/dropship/doublewide/buckle_mob(mob/living/buckling_mob, force = FALSE, check_loc = TRUE, lying_buckle = FALSE, hands_needed = 0, target_hands_needed = 0, silent)
+	if(chair_state == DROPSHIP_CHAIR_BROKEN)
+		balloon_alert_to_viewers("This chair is too damaged to sit in")
+		return FALSE
+	if(leader_chair && buckling_mob.skills.getRating(SKILL_LEADERSHIP) < SKILL_LEAD_TRAINED)
+		balloon_alert(buckling_mob, "You don't feel worthy enough to sit in this chair")
+		return FALSE
+	if(buckling_x)
+		src.pixel_x = buckling_x
+	return ..()
+
+/obj/structure/bed/chair/dropship/doublewide/post_buckle_mob(mob/buckling_mob)
+	buckling_mob.pixel_x = buckling_x
+	buckling_mob.old_x = buckling_x
+	doublewide_mob_density(buckling_mob, TRUE)
+	return ..()
+
+/obj/structure/bed/chair/dropship/doublewide/post_unbuckle_mob(mob/buckled_mob)
+	buckled_mob.pixel_x = initial(buckled_mob.pixel_x)
+	buckled_mob.old_x = initial(buckled_mob.pixel_x)
+	doublewide_mob_density(buckled_mob, FALSE)
+	return ..()
+
+/obj/structure/bed/chair/dropship/doublewide/proc/doublewide_mob_density(mob/interactionmob, buckling)
```

Comment:
doc your procs. Also why does this need to exist? Why not just handle this in each (un)buckle proc seperately?